File: PMV-App-Frontend/src/app/model/car.ts
import { Driver } from './driver';
import { ServiceTicket } from './serviceticket';

export class Car {
  plateNo!: string;
  carModel!: string;
  manufacturingYear!: number;
  upcomingServiceDate!: string;
  driver!: Driver;
  serviceTickets!: ServiceTicket[];
  isServiceDone!: boolean;

  constructor(){
    
  }


  // constructor(
  //   plateNo: string,
  //   carModel: string,
  //   manufacturingYear: number,
  //   upcomingServiceDate: string,
  //   driver: Driver,
  //   serviceTickets: ServiceTicket[],
  //   isServiceDone: boolean
  // ) {
  //   this.plateNo = plateNo;
  //   this.carModel = carModel;
  //   this.manufacturingYear = manufacturingYear;
  //   this.upcomingServiceDate = upcomingServiceDate;
  //   this.driver = driver;
  //   this.serviceTickets = serviceTickets;
  //   this.isServiceDone = isServiceDone;
  // }
}

File: PMV-App/src/main/java/com/app/pmvapp/model/Car.java
package com.app.pmvapp.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "car")
public class Car {

    @Id
    @Column(name = "plate_no")
    private String plateNo;

    @Column(name = "car_model")
    private String carModel;

    @Column(name = "manufacturing_year")
    private int manufacturingYear;

    @Column(name = "upcoming_service_date")
    private LocalDate upcomingServiceDate;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "driver_id")
    private Driver driver;

    @OneToMany(mappedBy = "car", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonIgnore
    private List<ServiceTicket> serviceTickets = new ArrayList<>();

    private Boolean isServiceDone = Boolean.FALSE;
}

